axum-extra/src/routing/spa.rs

```diff
@@ -147,7 +147,7 @@ impl<B, T, F> SpaRouter<B, T, F> {
     }
 }
 
-impl<B, F, T> From<SpaRouter<B, T, F>> for Router<B>
+impl<B, F, T> From<SpaRouter<B, T, F>> for Router<(), B>
```

Comment:
Is there a specific reason `SpaRouter` doesn't become generic as well?

Reply:
Yeah because it doesn't need to access the state and the user cannot add handlers directly to `SpaRouter`. I added a test in b6a2f39ca3dca just to make sure it actually works with routers that have state.